### PR TITLE
refactor: allow custom fonts to be passed in

### LINKED ISSUE #117: feat: use dynamic fonts based on languages
# Description

**Is this a BUG REPORT or FEATURE REQUEST ?** FEATURE

**What you expected to happen :**  

see : https://github.com/thematerials-network/CoA-schemas/issues/38

**How to reproduce it :** 

Use a certificate with `CN` CertificateLanguages.

- [ ] allow fonts to be dynamically loaded based on certificate languages

### LINKED ISSUE #150: refactor: PDF and HTML reports should use NotoSansSC
# Description

In order to support character sets other than latin in the PDFs, we will be switching to [NotoSans](https://fonts.google.com/noto/fonts) for the creation of reports. For now this would be [NotoSansSC](https://fonts.google.com/noto/specimen/Noto+Sans+SC?subset=chinese-simplified&query=noto) to support simplified chinese. The styles should be adjusted accordingly.

_NotoSans (right) uses more vertical space than Lato:_
image

## ToDo
- [ ] Adjust the report styles to accommodate the rendering of NotoSans in a more compact matter.
- [x] [Add section for Disclaimer](https://github.com/thematerials-network/CoA-schemas/issues/42)
- [x] [Reduce font size for Disclaimer](https://github.com/thematerials-network/CoA-schemas/issues/43)
- [x] Add "This is a computer-generated document. No signature is required."
- [x] Add link to CoA Website
- [x] Add Link to verification website
- [x] Update "Powered by S1Seven"

## Follow-up
- [x] [Adjust PDF rendering to use NotoSansSC as a default font.](https://github.com/thematerials-network/CoA-schemas/issues/38)
- [ ] Adjust default font of HTML rendering..

## Later work
- [ ] Facilitate rendering of different type sets like Japanese or Arabic.

### PR DESCRIPTION
**Pull request by eamon0989**
# Description

- Fixes #150 
- Fixes #117 

Refactor that allows more than 1 font to be used on a single line
Adds documentation examples for `generate-pdf`
Adds tests for `languageFontMap`
After refactoring how fixtures work (https://github.com/s1seven/schema-tools/issues/164), another fixture could be added with the Chinese font

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Refactor (a code change that neither fixes a bug nor adds a feature)
- [ ] Dependencies update
- [x] This change requires a documentation update

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by getlarge** [comment]
@eamon0989 Obviously i had a bad idea using the `node:` protocol in the imports for libraries that should be compatible with the browser as it made the Karma tests fail.
Could you revert that please ?

**Review by getlarge — CHANGES_REQUESTED** [review]

**Review by getlarge — CHANGES_REQUESTED** [review]

File: packages/generate-coa-pdf-template/test/builders.spec.ts [hunk 1]
```diff
@@ -126,6 +137,23 @@ describe('Rendering', () => {
...
+    const i18n = getI18N(translations, extraTranslations, ['EN', 'DE'], { DE: 'Helvetica' });
+    const receivers = createReceivers(certificate.Certificate.Parties as unknown as Parties, i18n);
+    const tableBody = receivers.table.body;
+    const titles = tableBody[0];
+    expect(tableBody[0].length).toEqual(3);
+    expect(titles[0][0]).toEqual(
+      expect.objectContaining({
+        text: [
+          { text: 'Customer / ', font: undefined },
+          { text: 'Kunde', font: 'Helvetica' },
```
[the remark this answers is not in the document]
Comment (eamon0989) [reply packages/generate-coa-pdf-template/test/builders.spec.ts:150]: I added it by default to the first one. Another possible option (not sure how feasible it is without looking into it) would be to detect when there are multiple objects, and automatically insert a `{ text: ' / ', font: undefined },` object between them?

[the remark this answers is not in the document]
Comment (eamon0989) [reply packages/generate-coa-pdf-template/test/builders.spec.ts]: I've updated this to add `{ text: ' / ', font: undefined },` when there are two translations present as it is a cleaner implementation.

File: packages/generate-pdf-template-helpers/src/helpers.ts [hunk 2]
```diff
@@ -49,6 +49,11 @@ export function computeTextStyle(
   locales?: string | string[],
 ): string | string[];
 export function computeTextStyle(value: string, format?: PossibleFormat, locales?: string | string[]): string;
+export function computeTextStyle(
```
Comment (eamon0989) [inline packages/generate-pdf-template-helpers/src/helpers.ts:52]: This is the only fix I could find for the compiler issue here: https://cloud.nx.app/runs/MURvaFXPBf
The types are compiled by `json-schema-to-typescript`.
The problem type is `[string, ...string[]]`.
Is there a better way to solve this problem? @getlarge

Comment (getlarge) [reply packages/generate-pdf-template-helpers/src/helpers.ts:52]: this are not really clean overloads...
Each overload should describe a specific input / output case plus one that contains all inputs / outputs unions.
```ts
// use this type for the function declaration and the most generic overload
type PossibleValues = string | string[] | [string, ...string[]];
export function computeTextStyle(
  value: PossibleValues,
  format?: PossibleFormat,
  locales?: string | string[],
): PossibleValues;

... all specific overloads in between

export function computeTextStyle(
  value: PossibleValues,
  format?: PossibleFormat,
  locales?: string | string[],
): PossibleValues {
...
}

File: packages/generate-pdf-template-helpers/src/translate.ts [hunk 3]
```diff
@@ -25,17 +36,26 @@ export class Translate<T = Translations, E = ExternalStandardsTranslations> {
...
-  ) {
-    return this.languages.map((language) => this.getField(language, group, phrase)).join(' / ');
+  ): TranslationWithFont[] {
+    const translationArray = this.languages.map((language) => {
+      const font = this.languageFontMap[language];
+      const field = this.getField(language, group, phrase);
+      return { text: field, font };
+    });
+    if (translationArray.length === 2) {
+      const middleObj = { text: ' / ', font: undefined };
```
Comment (getlarge) [inline packages/generate-pdf-template-helpers/src/translate.ts:46]:
```suggestion
     // maybe ?
      const separator = { text: ' / ', font: undefined };
```

File: packages/generate-pdf-template-helpers/src/translate.ts [hunk 4]
```diff
@@ -45,38 +65,71 @@ export class Translate<T = Translations, E = ExternalStandardsTranslations> {
...
+      const font = this.languageFontMap[language];
+      const text = this.getExtraField(externalStandard, language, propertyId, property) || defaultValue;
+      return { text, font };
+    });
 
-    return translatedFields[0] === translatedFields[1] ? translatedFields[0] : translatedFields.join(' / ');
+    if (
+      translatedFields[0]?.text === translatedFields[1]?.text ||
+      (translatedFields[0]?.text && !translatedFields[1]?.text)
+    ) {
```
Comment (getlarge) [inline packages/generate-pdf-template-helpers/src/translate.ts:86]: could you add a short comment about your intent above this condition ?

**Review by getlarge — CHANGES_REQUESTED** [review]

**Review by getlarge — APPROVED** [review]

**Pull request merged**